src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java

```diff
@@ -149,4 +193,101 @@ void validateCmekKey(FailureCollector failureCollector, Map<String, String> argu
     }
     validateCmekKeyLocation(cmekKeyName, null, location, failureCollector);
   }
+
+  public String getDataset() {
+    return dataset;
+  }
+
+  public String getDatasetProject() {
+    if (connection == null) {
+      return ServiceOptions.getDefaultProjectId();
+    }
+    return connection.getDatasetProject();
+  }
+
+  public String getProject() {
+    if (connection == null) {
+      throw new IllegalArgumentException(
+        "Could not get project information, connection should not be null!");
+    }
+    return connection.getProject();
+  }
+
+  @Nullable
+  public String tryGetProject() {
+    return connection == null ? null : connection.tryGetProject();
+  }
+
+  @Nullable
+  public String getServiceAccount() {
+    return connection == null ? null : connection.getServiceAccount();
+  }
+
+  @Nullable
+  public Boolean isServiceAccountFilePath() {
+    return connection == null ? null : connection.isServiceAccountFilePath();
+  }
+
+  @Nullable
+  public String getServiceAccountType() {
+    return connection == null ? null : connection.getServiceAccountType();
+  }
+
+  @Nullable
+  public String getBucket() {
+    if (bucket != null) {
```

Comment:
Are these methods just copied from the `BigQueryBaseConfig`? Can you open a JIRA to refactor these after we support connection in BQ pushdown?

Reply:
done, https://cdap.atlassian.net/browse/PLUGIN-977.